pkg/cluster/spec/spec.go

```diff
@@ -202,7 +202,12 @@ func (s *Specification) BaseTopo() *BaseTopo {
 func AllComponentNames() (roles []string) {
 	tp := &Specification{}
 	tp.IterComponent(func(c Component) {
-		roles = append(roles, c.Name())
+		switch c.Name() {
+		case ComponentTiSpark: // tispark-{master, worker}
```

Comment:
I think we should treat `tispark-{master,worker}` as a different component type.

Reply:
And we don't need `Role` method anymore.

Reply:
We need something called `tispark` to download the package